jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/BytecodeAnnotator.java

```diff
@@ -63,21 +89,63 @@ private static void LOG(boolean cond, String tag, String msg) {
   private static final String DIGEST_ENTRY_PATTERN =
       "Name: [A-Za-z0-9/\\$\\n\\s\\-\\.]+[A-Za-z0-9]\\nSHA-256-Digest: " + BASE64_PATTERN;
 
-  private static void addAnnotationIfNotPresent(
-      List<AnnotationNode> annotationList, String annotation) {
-    for (AnnotationNode node : annotationList) {
-      if (node.desc.equals(annotation)) {
-        return;
+  private static boolean annotationsShouldBeVisible(String nullableDesc) {
+    if (nullableDesc.equals(javaxNullableDesc)) {
+      return true;
+    } else if (nullableDesc.equals(androidNullableDesc)) {
+      return false;
```

Comment:
Just wanted to clarify - is this Android Nullable annotation always invisible?

Reply:
Yup. Invisible just means retention is `CLASS` as opposed to `RUNTIME`. The Android annotations are both `@Retention(RetentionPolicy.CLASS)`, at least all the versions IntelliJ brings up for our codebase.

Reply:
Okay. Sounds good.